Add diagnosis date propagation to synthetic breast cancer cohort gene…

adamblum opened the pull request:
…ration

also use dj_database_url for settings from DATABASE_URL

Commit:
Add diagnosis date propagation to synthetic breast cancer cohort generation

LeoMo42 commented on myenv/bin/Activate.ps1:
```diff
@@ -0,0 +1,241 @@
+<#
```
I don't think this should be checked in (the whole `myenv` content)

LeoMo42 commented on load_gibleed.sh:
```diff
@@ -0,0 +1,9 @@
+    # Convert file_name (e.g., PERSON) to PascalCase (e.g., Person)
+    model_name=$(echo "$file_name" | awk -F'_' '{for(i=1;i<=NF;i++){printf toupper(substr($i,1,1)) tolower(substr($i,2))}}')
+    echo "Loading $model_name from $f"
+    python manage.py load_csv_to_model "$model_name" "$f"
```
ok, `load_csv_to_model` in `/management` is good,

the `load_csv_to_model.py` in the root (same directory as this file) -- seems blank and useless

LeoMo42 commented on myenv/bin/activate:
```diff
@@ -0,0 +1,66 @@
+# This file must be used with "source bin/activate" *from bash*
```
same here and below (for all the `myenv` files)

LeoMo42 commented on omop/management/commands/load_patients.py:
```diff
@@ -0,0 +1,43 @@
+        }
+        yield info
+
+def insert_patient_info(info):
+    columns = ', '.join(info.keys())
+    values = ', '.join(['%s'] * len(info))
+    sql = f"INSERT INTO trials_patientinfo ({columns}) VALUES ({values})"
+    cursor.execute(sql, list(info.values()))
+    conn.commit()
```
Good to use Django models instead

```
PatientInfo.objects.create(**info)
```

LeoMo42 commented on omop/migrations/0002_cdm_core.py:
```diff
@@ -0,0 +1,310 @@
+# Generated migration: create OMOP CDM core tables for Eunomia GiBleed (v5.3)
+from django.db import migrations
+
+SQL_UP = r"""
```
hmmm, any idea WHY the vanilla SQL is used here?

did you use some options to force it?

And is it really needed here?

I thing the Django ORM-like instructions is more safe / more flexible

LeoMo42 commented on load_gibleed.sh:
```diff
@@ -0,0 +1,9 @@
+
+for f in ./data/GiBleed/GiBleed_5.3/CONDITION_OCCURRENCE.csv; do
```
the `CONDITION_OCCURRENCE` file is not checked in

if this is a public file, good to add a link to the file here

LeoMo42 reviewed (approved):
a few comments from me

overall -- looks good

Commit:
Add OHDSI Genomics extension tables to models

- Added Genomic variant table for storing genetic variants
- Added Genetic variant occurrence table for patient-specific variants
- Added Genomic biomarker table for biomarker data
- Added Molecular sequence table for sequence data
- Added Genomic relationship table for variant relationships
- All tables follow OHDSI CDM conventions with proper foreign keys and indexes
